source/nodejs/adaptivecards/src/host-config.ts

```diff
@@ -330,7 +330,166 @@ export class ContainerStyleSet {
     }
 }
 
+export class Version {
+    private _versionString: string;
+    private _major: number;
+    private _minor: number;
+    private _isValid: boolean = true;
+
+    constructor(major: number = 1, minor: number = 1) {
+        this._major = major;
+        this._minor = minor;
+    }
+
+    static parse(versionString: string): Version {
+        if (!versionString) {
+            return null;
+        }
+
+        var result = new Version();
+        result._versionString = versionString;
+
+        var regEx = /(\d+).(\d+)/gi;
+        var matches = regEx.exec(versionString);
+
+        if (matches != null && matches.length == 3) {
+            result._major = parseInt(matches[1]);
+
+            let minorVersion = matches[2];
+
+            while (minorVersion.length < 4) {
+                minorVersion += "0";
+            }
+
+            result._minor = Math.min(parseInt(minorVersion), 9999);
+        }
+        else {
+            result._isValid = false;
+        }
+
+        return result;
+    }
+
+    toString(): string {
+        let minorVersionString = this._minor.toString();
+
+        while (minorVersionString.length < 4) {
+            minorVersionString = "0" + minorVersionString;
+        }
+
+        let trailingZeros = 0;
+
+        for (let i = minorVersionString.length - 1; i >= 0; i--) {
+            if (minorVersionString[i] == "0") {
+                trailingZeros++;
+            }
+            else {
+                break;
+            }
+        }
+
+        minorVersionString = minorVersionString.substr(0, minorVersionString.length - trailingZeros);
+
+        return !this._isValid ? this._versionString : this._major + "." + minorVersionString;
+    }
+
+    compareTo(otherVersion: Version): number {
```

Comment:
> compareTo(otherVersion: Version): [](start = 3, length = 34)

looks like this won't work for wildcards or version ranges (per spec)

Reply:
My implementation doesn't support ranges at the moment, and uses the explicit string "any" for wildcard. There is an active discussion on the side about versioning. I think we should keep things simple, but I will update the code according to the conclusion of the conversation.

Reply:
Per discussion, we are going with a simple model for 1.2.